File: apps/Lodgr.Api/Models/Contract.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("contracts")]
public class Contract
{
    [Key]
    [Column("contract_id")]
    public long ContractId { get; set; }

    [Column("room_id")]
    public long RoomId { get; set; }

    [Column("tenant_id")]
    public long TenantId { get; set; }

    [Column("start_date")]
    public DateOnly StartDate { get; set; }

    [Column("end_date")]
    public DateOnly? EndDate { get; set; }

    [Column("actual_end_date")]
    public DateOnly? ActualEndDate { get; set; }

    [Column("monthly_rent", TypeName = "numeric(12,2)")]
    public decimal MonthlyRent { get; set; }

    [Range(typeof(decimal), "0", "79228162514264337593543950335")]
    [Column("deposit_amount", TypeName = "numeric(12,2)")]
    public decimal DepositAmount { get; set; }

    [Range(typeof(decimal), "0", "79228162514264337593543950335")]
    [Column("deposit_balance", TypeName = "numeric(12,2)")]
    public decimal DepositBalance { get; set; }

    [Column("deposit_status")]
    public DepositStatus DepositStatus { get; set; } = DepositStatus.UNPAID;

    [Range(1, int.MaxValue)]
    [Column("current_occupants")]
    public int CurrentOccupants { get; set; } = 1;

    [Range(1, 28)]
    [Column("billing_cycle_day")]
    public int BillingCycleDay { get; set; } = 1;

    [Column("status")]
    public ContractStatus Status { get; set; } = ContractStatus.DRAFT;

    [Column("note")]
    public string? Note { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }
}

public enum DepositStatus
{
    UNPAID,
    PARTIALLY_PAID,
    HELD,
    PARTIALLY_RETURNED,
    RETURNED,
    FORFEITED,
}

public enum ContractStatus
{
    DRAFT,
    ACTIVE,
    EXPIRED,
    TERMINATED,
}

File: apps/api/src/main/java/org/lodgr/api/model/Contract.java
package org.lodgr.api.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "contracts")
public class Contract {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "contract_id")
    private Long contractId;

    @Column(name = "room_id", nullable = false)
    private Long roomId;

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "actual_end_date")
    private LocalDate actualEndDate;

    @Column(name = "monthly_rent", nullable = false, precision = 12, scale = 2)
    private BigDecimal monthlyRent;

    @Column(name = "deposit_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal depositAmount;

    @Column(name = "deposit_balance", nullable = false, precision = 12, scale = 2)
    private BigDecimal depositBalance;

    @Enumerated(EnumType.STRING)
    @Column(name = "deposit_status", nullable = false)
    private DepositStatus depositStatus;

    @Column(name = "current_occupants", nullable = false)
    private Integer currentOccupants;

    @Column(name = "billing_cycle_day", nullable = false)
    private Integer billingCycleDay;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private ContractStatus status;

    @Column(name = "note")
    private String note;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    public Long getContractId() {
        return contractId;
    }

    public void setContractId(Long contractId) {
        this.contractId = contractId;
    }

    public Long getRoomId() {
        return roomId;
    }

    public void setRoomId(Long roomId) {
        this.roomId = roomId;
    }

    public Long getTenantId() {
        return tenantId;
    }

    public void setTenantId(Long tenantId) {
        this.tenantId = tenantId;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public LocalDate getActualEndDate() {
        return actualEndDate;
    }

    public void setActualEndDate(LocalDate actualEndDate) {
        this.actualEndDate = actualEndDate;
    }

    public BigDecimal getMonthlyRent() {
        return monthlyRent;
    }

    public void setMonthlyRent(BigDecimal monthlyRent) {
        this.monthlyRent = monthlyRent;
    }

    public BigDecimal getDepositAmount() {
        return depositAmount;
    }

    public void setDepositAmount(BigDecimal depositAmount) {
        this.depositAmount = depositAmount;
    }

    public BigDecimal getDepositBalance() {
        return depositBalance;
    }

    public void setDepositBalance(BigDecimal depositBalance) {
        this.depositBalance = depositBalance;
    }

    public DepositStatus getDepositStatus() {
        return depositStatus;
    }

    public void setDepositStatus(DepositStatus depositStatus) {
        this.depositStatus = depositStatus;
    }

    public Integer getCurrentOccupants() {
        return currentOccupants;
    }

    public void setCurrentOccupants(Integer currentOccupants) {
        this.currentOccupants = currentOccupants;
    }

    public Integer getBillingCycleDay() {
        return billingCycleDay;
    }

    public void setBillingCycleDay(Integer billingCycleDay) {
        this.billingCycleDay = billingCycleDay;
    }

    public ContractStatus getStatus() {
        return status;
    }

    public void setStatus(ContractStatus status) {
        this.status = status;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
